tests/30rooms/06invite.pl

```diff
@@ -2,39 +2,46 @@
 
 my $creator_preparer = local_user_preparer();
 
-my $inviteonly_room_preparer = preparer(
-   requires => [ $creator_preparer ],
+sub inviteonly_room_preparer
+{
+   my %args = @_;
 
-   do => sub {
-      my ( $user ) = @_;
-
-      matrix_create_room( $user,
-         # visibility: "private" actually means join_rule: "invite"
-         # See SPEC-74
-         visibility => "private",
-      )->then( sub {
-         my ( $room_id ) = @_;
-
-         do_request_json_for( $user,
-            method => "GET",
-            uri    => "/api/v1/rooms/$room_id/initialSync",
+   preparer(
+      requires => [ $args{creator} ],
+
+      do => sub {
+         my ( $creator ) = @_;
+
+         matrix_create_room( $creator,
+            # visibility: "private" actually means join_rule: "invite"
+            # See SPEC-74
+            visibility => "private",
          )->then( sub {
-            my ( $body ) = @_;
+            my ( $room_id ) = @_;
```

Comment:
I can't quite recall why all this checking logic is sat here. It probably doesn't want to live in the preparer itself; it should be done just once, in a test somewhere else. But leave it here for now since it was there before; I'll move it out sometime.